haicrypt/hcrypt_tx.c

```diff
@@ -52,20 +52,21 @@ int HaiCrypt_Tx_GetBuf(HaiCrypt_Handle hhc, size_t data_len, unsigned char **in_
 	return(crypto->msg_info->pfx_len);
 }
 
-int HaiCrypt_Tx_ManageKeys(HaiCrypt_Handle hhc, void *out_p[], size_t out_len_p[], int maxout) 
+int HaiCrypt_Tx_ManageKeys(HaiCrypt_Handle hhc, void *out_p[], size_t out_len_p[], int maxout)
 {
 	hcrypt_Session *crypto = (hcrypt_Session *)hhc;
-	hcrypt_Ctx *ctx = crypto->ctx;
-	int nbout = 0;
 
 	if ((NULL == crypto)
-	||  (NULL == ctx)
+	||  (NULL == crypto->ctx)
 	||  (NULL == out_p)
 	||  (NULL == out_len_p)) {
 		HCRYPT_LOG(LOG_ERR, "ManageKeys: invalid params: crypto=%p crypto->ctx=%p\n", crypto, ctx);
```

Comment:
crypto may be NULL here and dereference ctx

Reply:
No. The `NULL == crypto` condition is checked first, and if it's true, no other conditions are checked.

Reply:
`ctx` is not declared yet but used in the log.

Reply:
Right, missed that. Should be fixed now.